bundles/org.openhab.binding.tesla/src/main/java/org/openhab/binding/tesla/internal/handler/TeslaVehicleHandler.java

```diff
@@ -223,7 +223,11 @@ public void dispose() {
      * @return the vehicle id
      */
     public String getVehicleId() {
-        return vehicle.id;
+        if (vehicle != null) {
+            return vehicle.id;
+        } else {
+            return null;
```

Comment:
It looks valid as a bugfix. Just consider:
- If null is returned here, this is directly passed to `invokeAndParse()` as first parameter.
- `invokeAndParse()` handles this null parameter and returns null.
- When null it returned by `invokeAndParse()` this is also handled.

It works, but... Calling method will a null parameter just for that method to return null doesn't seem clean. At some point it must be considered an error and handled somehow, or new NPE's will likely be introduced. Not a request for change in context of this PR, just sharing thoughts.

Reply:
I agree it is not ideal, but the fact that this method accesses a field for returning its result is in general a bad design - as it has no influence over whether this is set or not. So returning `null` here in case it isn't (yet) initialized is the most logical thing to do witthout going into a more complex refactoring.

Reply:
I agree, more refactoring would be needed to improve this, so fully accepted as a bugfix. :)